pkg/storage/sqlcommon/sqlcommon.go

```diff
@@ -285,14 +325,22 @@ func Write(ctx context.Context, dbInfo *DBInfo, store string, deletes storage.De
 
 	changelogBuilder := dbInfo.stbl.
 		Insert("changelog").
-		Columns("store", "object_type", "object_id", "relation", "_user", "operation", "ulid", "inserted_at")
+		Columns(
+			"store", "object_type", "object_id", "relation", "_user",
+			"condition_name", "condition_context", "operation", "ulid", "inserted_at",
+		)
 
 	deleteBuilder := dbInfo.stbl.Delete("tuple")
 
 	for _, tk := range deletes {
 		id := ulid.MustNew(ulid.Timestamp(now), ulid.DefaultEntropy()).String()
 		objectType, objectID := tupleUtils.SplitObject(tk.GetObject())
 
+		conditionName, conditionContext, err := marshalRelationshipCondition(tk.GetCondition())
+		if err != nil {
+			return err
+		}
+
 		res, err := deleteBuilder.
 			Where(sq.Eq{
 				"store":       store,
```

Comment:
This new delete behavior brings some interesting nuances with it. What should be the expectations of the following:

```
Write(document:1#viewer@user:jon)
Delete(document:1#viewer@user:jon, {condition: "mycondition"})

Write(document:1#viewer@user:jon, {condition: "mycondition"})
Delete(document:1#viewer@user:jon)
```

If we still treat a tuple as `(object, relation, user)`, then the tuple keys that are to be deleted should be different than the tuple key that is to be inserted. That is,

```
Write({
    "writes": {
        "tuple_keys": [
            {"object": "document:1", "relation": "viewer", "user": "user:jon", "condition": {...}}
        ]
    },
    "deletes": {
        "tuple_keys": [
            {"object": "document:1", "relation": "viewer", "user": "user:jon"}
    }
})
```
The `condition` is not needed in the delete list.